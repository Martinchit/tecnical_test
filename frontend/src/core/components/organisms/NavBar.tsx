import React from 'react';
import { Link } from 'react-router-dom';
import { Navbar, Nav } from 'react-bootstrap';
import { Text } from '../atoms/Text';

interface Props {
  loggedIn: boolean;
  logOutAction: () => void;
}

export const NavBar: React.FC<Props> = ({ loggedIn, logOutAction }) => (
  <Navbar collapseOnSelect bg="dark" expand="lg" variant="dark">
    <Navbar.Brand as={Link} to="/">
      <Text fontSize={1.5} fontColor="#fff" fontWeight="bold">
        Execution System
      </Text>
    </Navbar.Brand>
    {loggedIn ? (
      <>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse
          id="responsive-navbar-nav"
          className="justify-content-end"
        >
          <Nav defaultActiveKey={window.location.href.split('/').reverse()[0]}>
            <Nav.Link as={Link} to="/stock_list" eventKey="stock_list">
              Stock Lists
            </Nav.Link>
            <Nav.Link as={Link} to="/orders_basket" eventKey="orders_basket">
              Orders Basket
            </Nav.Link>
          </Nav>
          <Nav.Link onClick={logOutAction}>Log Out</Nav.Link>
        </Navbar.Collapse>
      </>
    ) : null}
  </Navbar>
);
