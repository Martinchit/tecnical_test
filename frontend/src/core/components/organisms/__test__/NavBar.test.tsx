import React from 'react';
import { shallow } from 'enzyme';
import { Nav } from 'react-bootstrap';

import { NavBar } from '../NavBar';

describe('NavBar', () => {
  it('renders correctly', () => {
    const loggedIn = false;
    const logOutAction = jest.fn();
    const wrapper = shallow(
      <NavBar loggedIn={loggedIn} logOutAction={logOutAction} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Nav).length).toBe(0);
  });

  it('check current url', () => {
    const loggedIn = true;
    const logOutAction = jest.fn();
    const { location } = window;
    const href = 'http://test.com/test';
    delete window.location;
    window.location = {
      ...location,
      href,
    };
    const wrapper = shallow(
      <NavBar loggedIn={loggedIn} logOutAction={logOutAction} />
    );
    expect(wrapper.find(Nav).first().props().defaultActiveKey).toBe('test');
    window.location = location;
  });
});
