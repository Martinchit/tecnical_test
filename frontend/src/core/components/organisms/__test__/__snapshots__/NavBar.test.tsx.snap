// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<Navbar
  bg="dark"
  collapseOnSelect={true}
  expand="lg"
  variant="dark"
>
  <NavbarBrand
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Link",
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    to="/"
  >
    <styled.p
      fontColor="#fff"
      fontSize={1.5}
      fontWeight="bold"
    >
      Execution System
    </styled.p>
  </NavbarBrand>
</Navbar>
`;
